### ISSUE TITLE
Release v2.0.22 fails to run under Ubuntu 18.04

### ISSUE THREAD

**Issue by ksdd** [containers/conmon#225]
Release v2.0.22 fails to run under Ubuntu 18.04 with the following error message

> Error: failed to connect to container's attach socket: /run/user/205013812/libpod/tmp/socket/7c19123d148494b91f6689e0882d9f2b1b7b7bb8036128bbd4c363c11bbe3044/atta: dial unixpacket /run/user/205013812/libpod/tmp/socket/7c19123d148494b91f6689e0882d9f2b1b7b7bb8036128bbd4c363c11bbe3044/atta: connect: no such file or directory

I tested the binary provided by the [recommended repo](https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_18.04) as well as the [official release binary](https://github.com/containers/conmon/releases/download/v2.0.22/conmon-2.0.22.zip). Both fail to run with the same error.

BTW: When running the [official release binary](https://github.com/containers/conmon/releases/download/v2.0.22/conmon-2.0.22.zip) with --version option, I get:

> $ /usr/libexec/podman/conmon --version
> conmon version 2.0.23-dev
> commit: 715c680ed732f1e92166f03a7a94ff2f4c4f2f3e-dirty

Obviously the binary has been build from a dirty working copy and is **NOT** based on the commit 9c34a8663b85e479e0c083801e89a2b2835228ed  tagged as v2.0.22!

On the other hand the binary from the [recommended repo](https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_18.04) seems to be build based on v2.0.22 according to its version output.

**Comment by haircommander** [comment]
how are you running conmon? is this with podman or CRI-O?

**Comment by ksdd** [comment]
I run it via podman.

**Comment by haircommander** [comment]
what was the podman command you used? and what was the podman version?

**Comment by ksdd** [comment]
> $ podman --version
> podman version 2.2.1

> $ podman run --interactive --tty --rm alpine
> Error: failed to connect to container's attach socket: /run/user/205013812/libpod/tmp/socket/aa6712c41c455cd6c136c0af15d2345956ab9425e8690e4bd36845a59c384c83/atta: dial unixpacket /run/user/205013812/libpod/tmp/socket/aa6712c41c455cd6c136c0af15d2345956ab9425e8690e4bd36845a59c384c83/atta: connect: no such file or directory

**Comment by haircommander** [comment]
okay so I see now this was an unintended breaking change, and also easily reproducable. I will need to consult with the over developers on how best to handle it.

**Comment by haircommander** [comment]
thanks for your patience

**Comment by haircommander** [comment]
I've opened https://github.com/containers/conmon/pull/226 to fix

**Comment by ksdd** [comment]
Great! Thank You.

**Comment by ksdd** [comment]
Hi @haircommander,

today I've got an update of conmon to version 2.0.24 via [the mentioned repo](https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_18.04). I've seen, that #226 was merged before 2.0.24 was released. So I hoped, this issue is fixed. Unfortunately this is not the case. I get the same error as before. Are there still fixes needed or was it supposed to be fixed in 2.0.24?

**Comment by haircommander** [comment]
unfortunately, the  only way to preserve backward compatibility is to gate the new behaviour (that would fix your bug) behind a flag, meaning we also need a corresponding podman PR to fix your issue. that PR is here https://github.com/containers/podman/pull/8979 and I'd like to make it in before podman 3.0 release :)

**Comment by ksdd** [comment]
Okay, thank you. So I will wait for a new podman release and stick to the older conman version meanwhile.

**Comment by jnovy** [comment]
@ksdd Do you still see it? It seems to work for me:

```
$ rpm -q podman conmon
podman-5.4.1-1.fc41.x86_64
conmon-2.1.13-1.fc41.x86_64

```

```
$ podman run --interactive --tty --rm alpine
Resolved "alpine" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull docker.io/library/alpine:latest...
Getting image source signatures
Copying blob f18232174bc9 done   | 
Copying config aded1e1a5b done   | 
Writing manifest to image destination
/ # 
```

**Comment by ksdd** [comment]
No, I haven't seen it anymore. So I am going to close this issue.
